x-pack/plugins/security_solution/common/api/timeline/model/components.schema.yaml

```diff
@@ -301,15 +145,52 @@ components:
         - system_socket
         - threat_match
         - zeek
+    QueryMatchResult:
+      type: object
+      properties:
+        field:
+          type: string
+        displayField:
+          type: string
+        value:
+          type: string
+        displayValue:
+          type: string
+        operator:
+          type: string
+    DataProviderResult:
+      type: object
+      properties:
+        id:
+          type: string
+        name:
+          type: string
+        enabled:
+          type: boolean
+        excluded:
+          type: boolean
+        kqlQuery:
+          type: string
+        queryMatch:
+          $ref: '#/components/schemas/QueryMatchResult'
+        # and:
```

Comment:
@marshallmain thoughts on how to handle circular deps? Not sure if the generator can handle: https://zod.dev/?id=recursive-types ?